src/org/etools/j1939_84/controllers/part03/Part03Step16Controller.java

```diff
@@ -54,12 +58,41 @@
     @Override
     protected void run() throws Throwable {
         // 6.3.16.1.a. Turn the engine off.
+        incrementProgress("Part 3, Step 16 Turn Engine Off and keep the ignition key in the off position");
+        ensureKeyOffEngineOff();
+
         // 6.3.16.1.b. Confirm Fault A is still implanted according to the manufacturer’s instruction.
+        incrementProgress("Waiting for implant of Fault A according to the engine manufacturer's instruction");
+        waitForFault("Part 6.3.16.1.b");
+
         // 6.3.16.1.c. Wait manufacturer’s recommended interval with the key in the off position.
+        incrementProgress("Waiting for manufacturer's recommended interval after the Fault A implant");
+        waitForManufacturerInterval();
+
         // 6.3.16.1.d. Turn ignition key to the ON position.
         // 6.3.16.1.e. Observe MIL and Wait to Start Lamp in Instrument Cluster
         // 6.3.16.1.f. Start Engine after MIL and Wait to Start Lamp (if equipped) have extinguished.
         // 6.3.16.1.g. Wait as indicated by the engine manufacturer’s recommendations for Fault A.
+        waitForEngineStart();
+    }
+
+    private void waitForManufacturerInterval() {
+        String message = "Please wait for the manufacturer's recommended interval with the key in off position" + NL;
+        message += "Press OK to continue the testing" + NL;
+        if (!isDevEnv()) {
+            displayInstructionAndWait(message, "Part 6.3.16.1.c", WARNING);
+        }
+    }
+
+    private void waitForEngineStart() {
+        String message = "Turn ignition key to the ON position" + NL;
+        message += "Please observe the MIL and Wait to Start Lamp (if equipped) in the Instrument Cluster" + NL;
+        message += "Start Engine after MIL and Wait to Start Lamp (if equipped) have extinguished" + NL;
+        message += "Please wait as indicated by the engine manufacturer’s recommendations for Fault A" + NL;
+        message += "Press OK when ready to continue testing" + NL;
+        if (!isDevEnv()) {
```

Comment:
nit: this `isDevEnv()` check can happen before the `String` is created as the string will never be shown so it doesn't need to be created.  You might even be able to move the check up higher.